src/main/java/io/github/zap/commons/event/SimpleEvent.java

```diff
@@ -0,0 +1,253 @@
+ *
+ * The size of the internal array scales similarly to that of {@link ArrayList}, except you can specify the amount by
+ * which the array size is increased per resize in the constructor. The array is only resized when it is necessary to
+ * do so to fit a new handler. The default increment is 8.
+ *
+ * SimpleEvent instances catch all exceptions thrown by any registered handlers. Caught exceptions may only be
+ * "handled" by user code after all events have been called. This is done by supplying a custom instance of
+ * {@link ExceptionHandler} in the constructor (see {@link ExceptionHandlers} for default factory methods). The
+ * handler will be passed the <i>first</i> {@link Throwable} instance that was produced, if any, and subsequent
+ * Throwables will be suppressed (see {@link Throwable#addSuppressed(Throwable)}). The default exception handler will
+ * be used if none is supplied, which is to simply log exceptions using the global logger.
+ *
+ * This class is not thread safe. It does not support recursively invoking handlers (that is, calling the
+ * {@link SimpleEvent#invoke(Object)} method while inside a handler that is currently being invoked by this instance).
+ * Attempting to do so will result in an unchecked {@link IllegalStateException}.
+ * @param <T> The type of argument handlers will receive
+ */
+@SuppressWarnings({"rawtypes", "unchecked"})
+public class SimpleEvent<T> implements Event<T> {
+    private static final int DEFAULT_EXCESS_CAPACITY = 8;
+
+    private class HandlerModification {
+        private final boolean isAdd;
+        private final EventHandler<T> handler;
+
+        private HandlerModification(boolean isAdd, EventHandler<T> handler) {
+            this.isAdd = isAdd;
+            this.handler = handler;
+        }
+    }
+
+    private EventHandler[] bakedHandlers;
+
+    private final ExceptionHandler exceptionHandler;
+    private final int excessCapacity;
+    private final List<HandlerModification> modifications = new LinkedList<>();
+
+    private int size = 0;
+    private boolean invoking = false;
+    private boolean clearFlag = false;
+    private boolean rebuildFlag = false;
+
+    public SimpleEvent(@NotNull ExceptionHandler exceptionHandler, int excessCapacity) {
+        Validate.isTrue(excessCapacity >= 0, "excessCapacity cannot be negative");
+
+        this.exceptionHandler = Objects.requireNonNull(exceptionHandler, "exceptionHandler cannot be null");
+        this.excessCapacity = excessCapacity;
+        this.bakedHandlers = new EventHandler[excessCapacity];
+    }
+
+    public SimpleEvent(@NotNull ExceptionHandler exceptionHandler) {
+        this(exceptionHandler, DEFAULT_EXCESS_CAPACITY);
+    }
+
+    public SimpleEvent(@NotNull Logger logger) {
+        this(ExceptionHandlers.logHandler(logger));
+    }
+
+    public SimpleEvent() {
+        this(ExceptionHandlers.logHandler(Logger.getGlobal()));
+    }
```

Comment:
IMO, the default `ExceptionHander` should be `ExceptionHanders.rethrow()` because inexperienced developers will have some wtf moment when debugging exception

Reply:
sure